### PR TITLE
adding CI checks

### LINKED ISSUE #199: Show CI status in PRs
Setup continuous integration to show status (compiling, tests) in PRs.

### PR DESCRIPTION
**Pull request by mellowgeek**
adding workflow files for CI checks on pull request and push to develop. 
Added NDK version to the build.gradle file because of a bug in the Android Gradle Plugin that causes the GitHub Action CI check to fail without this. Link to issue [https://issuetracker.google.com/issues/144111441](https://issuetracker.google.com/issues/144111441)

### CODE REVIEW

File: app/src/main/java/org/coepi/android/worker/tcnfetcher/ContactsFetchManager.kt [hunk 1]
```diff
@@ -22,7 +22,9 @@ class ContactsFetchManager(context: Context) {
             .setRequiredNetworkType(CONNECTED)
             .build()
 
-        return Builder(ContactsFetchWorker::class.java, 15, MINUTES)
+        val interval : Long = 15
```
Comment (ivnsch) [inline app/src/main/java/org/coepi/android/worker/tcnfetcher/ContactsFetchManager.kt]: Was it needed to declare the type as `Long`? 🤔

**Comment by ivnsch** [comment]
Can you add links to the Github issue and bug report to the description of the PR, and link to the PR in the code, please?

**Comment by ivnsch** [comment]
Also, the CI not running the tests?

File: app/src/main/java/org/coepi/android/worker/tcnfetcher/ContactsFetchManager.kt (hunk 1 above)
Comment (mellowgeek) [reply app/src/main/java/org/coepi/android/worker/tcnfetcher/ContactsFetchManager.kt]: Not declaring it was causing a lint error.

```
Errors found:
  
  /home/runner/work/app-android/app-android/app/src/main/java/org/coepi/android/worker/tcnfetcher/ContactsFetchManager.kt: Error: Unexpected failure during lint analysis of ContactsFetchManager.kt (this is a bug in lint or one of the libraries it depends on)
  
  Message: java.lang.Integer cannot be cast to java.lang.Long
  
  The crash seems to involve the detector androidx.work.lint.InvalidPeriodicWorkRequestIntervalDetector.
  You can try disabling it with something like this:
      android {
          lintOptions {
              disable "InvalidPeriodicWorkRequestInterval"
          }
      }
```

Comment (ivnsch) [reply app/src/main/java/org/coepi/android/worker/tcnfetcher/ContactsFetchManager.kt]: Okay. Did you try adding `L` at the end of the number?

Comment (mellowgeek) [reply app/src/main/java/org/coepi/android/worker/tcnfetcher/ContactsFetchManager.kt]: I did not, fixed.

**Comment by mellowgeek** [comment]
Added unit tests as a check, results are uploaded as a build artifact.

**Comment by ivnsch** [comment]
Can you add the instrumentation tests too? We use these for JNI.

**Review by ivnsch — APPROVED** [review]
The configuration seems to work now. I wonder if we can execute the JNI tests as Unit Tests, because 7 minutes is a lot of time for this. But we can investigate it in a separate issue.

**Pull request merged**